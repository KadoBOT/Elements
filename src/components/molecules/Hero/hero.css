.hero {
  background: url('../../../assets/Blog/device-lab-testing.jpg');
  height: 39vw;
  background-size: cover;
  background-attachment: fixed;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 3.5vw;
  z-index: 0;
  text-shadow: 1px 1px 2px rgba(0, 0, 0, 0.58);
}

.hero__gradient {
  background: -moz-linear-gradient(left,  rgba(0,0,0,0.65) 0%, rgba(0,0,0,0.59) 45%, rgba(0,0,0,0) 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(left,  rgba(0,0,0,0.65) 0%,rgba(0,0,0,0.59) 45%,rgba(0,0,0,0) 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to right,  rgba(0,0,0,0.65) 0%,rgba(0,0,0,0.59) 45%,rgba(0,0,0,0) 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#a6000000', endColorstr='#00000000',GradientType=1 ); /* IE6-9 */

  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  z-index: -1;
}
