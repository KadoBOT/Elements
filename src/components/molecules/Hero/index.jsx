import React, {PropTypes} from 'react'

import { HeroTitle, HeroSubtitle } from '../../atoms/HeroText'
import './hero.css'

const Hero = ({overlay, title, subtitle}) => {
  return(
    <div className="hero">
      {overlay && <div className="hero__gradient" />}
      <HeroTitle>{title}</HeroTitle>
      <HeroSubtitle>{subtitle}</HeroSubtitle>
    </div>
  )
}

Hero.propTypes = {
  props: PropTypes.type
}

export default Hero
