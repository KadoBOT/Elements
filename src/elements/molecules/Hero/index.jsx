import React, {PropTypes} from 'react'

import { HeroTitle, HeroSubtitle } from '../../atoms/HeroText'
import './hero.css'

const Hero = ({overlay, title, subtitle}) => (
  <div className="hero">
    {overlay && <div className="hero__gradient" />}
    <HeroTitle>{title}</HeroTitle>
    <HeroSubtitle>{subtitle}</HeroSubtitle>
  </div>
)

Hero.propTypes = {
  overlay: PropTypes.bool,
  title:PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
}

export default Hero
