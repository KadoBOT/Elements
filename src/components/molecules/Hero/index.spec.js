import React from 'react'
import { shallow, render, mount } from 'enzyme';

import Hero from './'

it('should render the Hero', () => {
    const wrapper = shallow(
        <Hero overlay={true} title="1" subtitle="2" />
    );
    expect(wrapper).toMatchSnapshot();
});
