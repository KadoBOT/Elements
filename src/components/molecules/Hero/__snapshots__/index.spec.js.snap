exports[`test should render the Hero 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="hero">
    <div
        className="hero__gradient" />
    <styled.p>
        1
    </styled.p>
    <styled.p>
        2
    </styled.p>
</div>,
  "nodes": Array [
    <div
      className="hero">
      <div
            className="hero__gradient" />
      <styled.p>
            1
      </styled.p>
      <styled.p>
            2
      </styled.p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Hero
        overlay={true}
        subtitle="2"
        title="1" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "overlay": true,
          "subtitle": "2",
          "title": "1",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="hero">
          <div
                    className="hero__gradient" />
          <styled.p>
                    1
          </styled.p>
          <styled.p>
                    2
          </styled.p>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="hero">
          <div
                    className="hero__gradient" />
          <styled.p>
                    1
          </styled.p>
          <styled.p>
                    2
          </styled.p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Hero
    overlay={true}
    subtitle="2"
    title="1" />,
}
`;
